internal/config/attributemap.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/uber-go/mapdecode"
+)
+
+// AttributeMap is a convenience type on top of a map
+// that gives us a cleaner interface to validate config values.
+type AttributeMap map[string]interface{}
+
+// PopString will pop a value from the attribute map and return the string
+// it points to, or an error if it couldn't pop the value and decode.
+func (m AttributeMap) PopString(name string) (s string, err error) {
+	_, err = m.Pop(name, &s)
+	return
```

Comment:
In general, this style isn't encouraged (I'd have to find the right blog posts but yea). If anything, it's inconsistent, I'd just do:

```go
var s string
_, err := m.Pop(name, &s)
return s, err
```

Same for the rest of the code here.